source/common/router/header_parser.cc

```diff
@@ -16,6 +16,13 @@
 namespace Envoy {
```

Comment:
I think this should be moved also as it's part of the formatter?

Reply:
I attempted this and I'm running into weird namespace problems, seemingly because the HeaderParserPtr is used within the HeaderParser class, and the forward-declare seems to break classes out-of-namespace from seeing it. I don't really have time to dig into this further, I think I'd rather the rest of these moves be handled by someone who has more context around the formatter header stuff and plans and has plans to refactor this.

Reply:
If we aren't going to finish the job let's just abandon the PR? I would rather not leave this half done when this is an aesthetic cleanup anyway. Thank you!

/wait